main.go

```diff
@@ -92,24 +113,41 @@ func UpgradeResponderCmd() cli.Command {
 }
 
 func startUpgradeResponder(c *cli.Context) error {
-	cfg := c.String(FlagUpgradeResponseConfiguration)
-	if cfg == "" {
-		return fmt.Errorf("no upgrade response configuration specified")
+	if err := validateCommandLineArguments(c); err != nil {
+		return err
 	}
 
+	cfg := c.String(FlagUpgradeResponseConfiguration)
 	influxURL := c.String(FlagInfluxDBURL)
 	influxUser := c.String(FlagInfluxDBUser)
 	influxPass := c.String(FlagInfluxDBPass)
+	influxContinuousQueryPeriod := c.String(FlagInfluxDBContinuousQueryPeriod)
+	applicationName := c.String(FlagApplicationName)
 	geodb := c.String(FlagGeoDB)
+	port := c.String(FlagPort)
+
 	done := make(chan struct{})
-	server, err := upgraderesponder.NewServer(done, cfg, influxURL, influxUser, influxPass, geodb)
+	server, err := upgraderesponder.NewServer(done, applicationName, cfg, influxURL, influxUser, influxPass, influxContinuousQueryPeriod, geodb)
 	if err != nil {
 		return err
 	}
 	router := http.Handler(upgraderesponder.NewRouter(server))
 
-	listen := "0.0.0.0:8314"
-	go http.ListenAndServe(listen, router)
+	listeningAddress := "0.0.0.0"
+	if port != "" {
+		listeningAddress = listeningAddress + ":" + port
```

Comment:
use a default value in cli instead.  https://github.com/urfave/cli/blob/master/docs/v2/manual.md